pkg/runtime/wfengine/backendFactory.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package wfengine
+
+import (
+	"time"
+
+	"github.com/dapr/dapr/pkg/runtime/processor/workflowBackend"
+	"github.com/dapr/kit/logger"
+	"github.com/microsoft/durabletask-go/backend"
+	"github.com/microsoft/durabletask-go/backend/sqlite"
+)
+
+const (
+	ActorBackendType  = "workflowbackend.actor"
+	SqliteBackendType = "workflowbackend.sqlite"
+)
+
+type BackendFactory func(appId string, wfe *WorkflowEngine, backendComponentInfo *workflowBackend.WorkflowBackendComponentInfo, log logger.Logger) backend.Backend
+
+var backendFactories = map[string]BackendFactory{
+	ActorBackendType:  getActorBackend,
+	SqliteBackendType: getSqliteBackend,
+}
+
+func getActorBackend(appId string, wfe *WorkflowEngine, backendComponentInfo *workflowBackend.WorkflowBackendComponentInfo, log logger.Logger) backend.Backend {
+	wfe.BackendType = ActorBackendType
+	return NewActorBackend(appId, wfe)
+}
+
+func getSqliteBackend(appId string, wfe *WorkflowEngine, backendComponentInfo *workflowBackend.WorkflowBackendComponentInfo,
+	log logger.Logger) backend.Backend {
+	wfe.BackendType = SqliteBackendType
+	var sqliteOptions = &sqlite.SqliteOptions{}
+
+	if connectionString, ok := backendComponentInfo.WorkflowBackendMetadata.Properties["connectionString"]; ok {
+		sqliteOptions.FilePath = connectionString
+	}
+
+	if orchestrationLockTimeout, ok := backendComponentInfo.WorkflowBackendMetadata.Properties["orchestrationLockTimeout"]; ok {
+		orchestrationLockTimeout, err := time.ParseDuration(orchestrationLockTimeout)
+		sqliteOptions.OrchestrationLockTimeout = orchestrationLockTimeout
+
+		if err != nil {
+			sqliteOptions.OrchestrationLockTimeout = orchestrationLockTimeout
+		} else {
+			log.Errorf("invalid orchestrationLockTimeout provided in backend workflow component: %v", err)
```

Comment:
It looks like the error handling logic is reversed here and further below.

Reply:
Fixed.